server/channels/api4/group.go

```diff
@@ -641,7 +641,7 @@ func unlinkGroupSyncable(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	c.App.Srv().Go(func() {
-		c.App.SyncRolesAndMembership(c.AppContext, syncableID, syncableType, c.Params.GroupId)
+		c.App.RemoveMembershipsFromUnlinkedSyncable(c.AppContext, syncableID, syncableType)
```

Comment:
I'm still getting a handle on the rules -- if we unlink a channel from a group, do we force remove memberships? All the group's members?

Reply:
We only force removal of members from a channel/team if the channel/team is group constrained. The query will check if the given syncable is group constrained. 

You made me think of an annoying race condition with my solution though because we are now linking/unlinking first and then updating the channel `group_constrained` status. So if they unlink a group and change the channel to no longer be `group_constrained` then the users still get removed from the channel even though they shouldn't. 

I guess it makes sense to update the order of requests again to go `link`->`patch channel`->`unlink`. It's currently `link/unlink`->`patch channel`.

Reply:
Nothing in particular, just wanted to make sure we've thought of all the combinations.